### PR TITLE
WD-253: Introduce JIRA ticket number validation in commit messages.

### PR DESCRIPTION
**Pull request by tormi**
Ticket: WD-253

## Changes

This PR introduces JIRA ticket number validation in GIT commit messages via GrumPHP [git_commit_message](https://github.com/phpro/grumphp/blob/v2.x/doc/tasks/git_commit_message.md) task.

General [JIRA ticket code regex](https://stackoverflow.com/a/73914895) is implemented as a default rule: `/^[A-Z][A-Z0-9_]+-[1-9][0-9]*: /`.

When implementing in a specific project, the name of the JIRA project should be specified. For example, the regex for a project called WD would be: `/^WD-[1-9][0-9]*: /`.

## Testing

- Try making commits without a JIRA ticket reference, it should fail.
- Try making commits with JIRA ticket name at the beginning of commit message, f. ex: `LA004311-3325: `. It should pass.

### CODE REVIEW

**Review by misterjoonas — CHANGES_REQUESTED** [review]
This would not work for example in a project where a ticket ID can be LA004311-3325. See https://stackoverflow.com/a/30518972 for Jira regex.

**Comment by tormi** [comment]
@badrange, good point when considering this particular template project!

File: grumphp.yml [hunk 1]
```diff
@@ -7,6 +7,13 @@ grumphp:
     failed: ~
     succeeded: ~
   tasks:
+    git_commit_message:
+      max_body_width: 0
+      max_subject_width: 0
+      matchers:
+        # Modify the universal JIRA regex provided to suit your project requirements, incl. name.
+        # For example, the regex for a project called WD would be: `/^WD-[1-9][0-9]*: /`.
+        'The commit must start with the JIRA issue number': '/^[A-Z][A-Z0-9_]+-[1-9][0-9]*: /'
```
[the remark this answers is not in the document]
Comment (tormi) [reply grumphp.yml:16]: Let's continue discussions on this topic. I'm not too sure that allowing such a back hole is a great idea. This actually defeats the purpose of adding a ticket number rule.

[the remark this answers is not in the document]
Comment (tormi) [reply grumphp.yml:16]: Regex should validate project-specific ticket numbers, so it should be configured on a project-by-project basis. But the values proposed in the template should be purposeful.

**Comment by tormi** [comment]
> Please consider how this change will affect possible outside contributors who don't have access to your internal JIRA

@badrange - It looks like we can use the `GH-<issue number>` syntax for this, similar to issue numbers in JIRA, see [Issues and Pull Requests](https://docs.github.com/en/get-start/writing-on- github/working-with-advanced-formatting/autolinked-references-and-urls#issues-and-pull-requests):

```md
GH- and issue or pull request number	GH-26	[GH-26](https://github.com/jlord/sheetsee.js/issues/26)
```

The same regex applies as for JIRA issues. I'll include some instructions on this in the readme.

**Comment by tormi** [comment]
Thanks, @misterjoonas & @badrange!

**Pull request merged**